File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    stack = []
    for i in s:
        if i == '(':
            stack.append(i)
            continue
        if len(stack) == 0 or stack.pop() != '(':
            return False
    
    if len(stack):
        return False

    return True

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = []
    for (const i of s) {
        if (i === '(') {
            stack.push(i)
            continue
        }
        if (stack.length === 0 || stack.pop() !== '(') return false
    }
    if (stack.length) return false   
    
    return true;
}